core/queriec.c

```diff
@@ -0,0 +1,79 @@
+#include <stdio.h>
+#include <stddef.h>
+#include <stdarg.h>
+
+#include "queriec.h"
+
+void
+queriec_init(struct queriec_obj *obj, size_t size, size_t offset)
+{
+    obj->size = size;
+    obj->status = 0;
+    obj->offset = offset;
+}
+
+void
+_queriec_add(struct queriec_obj *obj, char *query, char *letter, char key[], size_t keySize, char value[], size_t valueSize)
+{
+    size_t idx = obj->offset, i = 0;
+        
+    query[idx++] = letter[0];
+    while (1) {
+        if (i < keySize) query[idx++] = key[i++];
+        else {
+            idx--;
+            break;
+        }
+    }
+    i = 0;
+    query[idx++] = '=';
+    while (1) {
+        if (i < valueSize) query[idx++] = value[i++];
+        else break;
+    }
```

Comment:
```suggestion
    while (i < valueSize) {
        query[idx++] = value[i++];
    }
```
Would this accomplish the same thing?